rfcs/0000-windows-containers.md

```diff
@@ -0,0 +1,119 @@
+## Problem:
+
+kpack does not support building containers that can run on windows
+
+## Outcomes:
+
+### kpack should run on clusters that have windows workers
+
+kpack service pods and build pods for linux images will need to know not to run on the windows workers to work in a mixed-os cluster.
+
+The kpack release configuration should specify to run pods explicitly on linux os.
+The kpack controller should schedule linux builds on linux workers.
+
+**Actions to Take**
+
+Use the following node selector for all service pods and build pods for linux images:
+
+```
+nodeSelector:
+    kubernetes.io/os: linux
+```
+
+### kpack should support creating windows builders
+
+This is currently supported in the [pack cli](https://github.com/buildpacks/pack/issues/469)
+
+**Actions to Take**
+
+- kpack-controller will use a new environment variable `WINDOWS_LIFECYCLE_IMAGE` to reference the windows lifecycle image
+- When a Builder or ClusterBuilder is created using a windows stack and windows buildpackages, kpack-controller must create a windows builder using the windows lifecycle image.
+    - Image os is determined by the image [os property](https://github.com/opencontainers/image-spec/blob/master/config.md#:~:text=os%20string).
+
+Notes:
+
+- Builders with mixed os images must error gracefully
+- Builders should be labeled with `os.kpack.io: windows` for os disambiguation independent of image metadata
+
+### kpack should support running builds on windows workers
+
+**Actions to Take**
+
+- kpack-controller will accept new environment variables for windows images needed to run builds. These will have the same functionality as on linux and will be used for running builds on windows workers.
+    - `WINDOWS_BUILD_INIT_IMAGE` - a windows build init image
```

Comment:
This is creating a combinatorial explosion. Perhaps, we should just have one windows image that can build-init/complete/rebase?

Reply:
I think that's viable

I would prefer the windows behavior mirror the linux for consistency and maintainability

Reply:
It might make sense to do this for linux as well.

Reply:
If you all are considering ARM support too, that may add yet another dimension. Docker hub images rely heavily on [image indexes](https://github.com/opencontainers/image-spec/blob/master/image-index.md) (aka manifest lists) currently, for which k8s appears to have [patterns and support](https://kubernetes.io/docs/concepts/containers/images/#multi-architecture-images-with-manifests). Also, ggcr lets you filter indexes with [`remote.WithPlatform()`](https://github.com/google/go-containerregistry/blob/a84993334b277818146f48584a1ad748e1b53859/pkg/v1/remote/options.go#L114).

Reply:
I think ARCH concerns is out of scope for the rfc